wasm/src/main/java/com/dylibso/chicory/wasm/Parser.java

```diff
@@ -512,67 +506,85 @@ private static ElementSection parseElementSection(ByteBuffer buffer, long sectio
     }
 
     private static Element parseSingleElement(ByteBuffer buffer) {
-        var kind = readVarUInt32(buffer);
-        switch ((int) kind) {
```

Comment:
The specification defines three kinds of element. They are read in in various ways. The integer is interpreted as a bit field which defines the form of the element.

See this note in the spec:

```quote
Note: The initial integer can be interpreted as a bitfield. Bit 0 indicates a passive or declarative segment, bit 1 indicates the presence of an explicit table index for an active segment and otherwise distinguishes passive from declarative segments, bit 2 indicates the use of element type and element [expressions](https://webassembly.github.io/spec/core/binary/instructions.html#binary-expr) instead of element kind and element indices.

Additional element kinds may be added in future versions of WebAssembly.
```

Reply:
Also have a look here, where the semantic meaning of elements is described: https://webassembly.github.io/spec/core/syntax/modules.html#element-segments

Reply:
Note specifically that the element type does *not in any way* correspond to the sections of a WASM file, as the previous code seemed to presume.